src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,22 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        protected virtual void BuildAfterExecuteContent(CSharpCodeWriter writer, IList<Chunk> chunks)
```

Comment:
`/// docs` here and for `DecorateChunks()` since the next dev (e.g. in MVC repo) won't know what's up. I know this class has few doc comments but it's something we already use in other repos.